c_src/dpiConn_nif.c

```diff
@@ -287,9 +287,9 @@ DPI_NIF_FUN(conn_close)
 
     if (!enif_get_resource(env, argv[0], dpiConn_type, (void **)&connRes))
         BADARG_EXCEPTION(0, "resource connection");
-
-    if (!enif_is_list(env, argv[1]) &&
-        !enif_get_list_cell(env, argv[1], &head, &tail))
+    if (enif_is_list(env, argv[1]))
+        enif_get_list_cell(env, argv[1], &head, &tail);
+    else
```

Comment:
As per the conditions:
![image](https://user-images.githubusercontent.com/913008/60450330-f0db3200-9c29-11e9-885b-1c33a4d96432.png)
and from
![image](https://user-images.githubusercontent.com/913008/60454005-04d76180-9c33-11e9-8de1-f5903acd53a0.png)

Please check if the rewrite works
```c
unsigned len;
if (!enif_get_list_length(env, argv[1], &len))
    BADARG_EXCEPTION(1, "atom list modes, not a list");
if (len > 0 && !enif_get_list_cell(env, argv[1], &head, &tail))
    BADARG_EXCEPTION(1, "atom list modes");
```

Reply:
yes, that also works